src/Propel/Generator/Behavior/Sluggable/SluggableBehavior.php

```diff
@@ -291,7 +291,7 @@ protected function makeSlugUnique(\$slug, \$separator = '" . $this->getParameter
             $script .= "
 
         \$count = " . $this->builder->getStubQueryBuilder()->getClassname() . "::create()
-            ->filterBySlug(\$this->$getter())
+            ->filterBySlug(\$slug)
```

Comment:
Yeah. 299 would make perfect sense also.
Not sure about 317. And there's one more that I think should not be changed as well.

The only thing I can think is that this might interfere if you're trying to explicitly change the slug using setSlug()

I will investigate further.